data/StoreRecord.ts

```diff
@@ -147,6 +148,24 @@ export class StoreRecord {
         return this.validator?.validationState ?? 'Unknown';
     }
 
+    get modifiedData(): PlainObject {
+        if (!this.isModified) return {};
+
+        const ret = this.fields.reduce((acc, {name}) => {
+            if (isNil(this.committedData) || this.data[name] !== this.committedData[name]) {
+                acc[name] = this.data[name];
+            }
+            return acc;
+        }, {});
+
+        return isEmpty(ret)
```

Comment:
normalize empty here to null.  Still on the fence about including the id, but I guess that reduces the burden on the calling code.